core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/SearchResourceRequest.java

```diff
@@ -344,9 +334,10 @@ public final B execute(C context) {
 	@VisibleForTesting
 	static Options processSpecialOptionKey(Options options, Enum<?> specialOptionKey) {
 		if (specialOptionKey != null && options.containsKey(specialOptionKey)) {
-			// this will throw a CCE if non-String value is encountered in the option key and that is okay
-			String specialOption = options.getString(specialOptionKey);
-			if (specialOption.startsWith(SPECIAL_OPTION_CHARACTER) && specialOption.endsWith(")")) {
+			// this will throw a CCE if non-TermFilter value is encountered in the option key and that is okay
+			final TermFilter termFilter = options.get(specialOptionKey, TermFilter.class);
+			String specialOption = termFilter.getTerm();
```

Comment:
Please keep the original feature intact and add an extractor to extract the term value for the special option key handler.